### DISCUSSION TITLE
How the  "PnP - filter Search" web part show all the options inside a ComboxBox

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#2718] [Q&A]
I added a Filter search web part, which have a single filter field based on the FileType of type Combo-box. as follow:-

![image](https://user-images.githubusercontent.com/79726511/214754170-daac8a05-e053-4e7f-9e0c-60eb467065fe.png)

now my question is how the Filter's ComboBox options will be populated? let say we have 3 million documents. and our Search Result web part will show all the 3 million documents (100 items per page). so will the FileType Combo-box contain the unique file types for all the 3 million documents? or it will only show the options for the search result currently retrieved from the server? i am not sure how the search results will be shown? will it get loaded in patches or not.. but i can not understand how the Filter field combo-box options are been populated? 

Let take this as an example. let say we have only one .xls document out of the 3 million documents. and let say this .xls document will be the  last document shown  inside the Search Result, So will this option ".xls" be shown when the PnP Filter search web part got rendered for the first time, even if this .xls file have not been loaded from the server yet? 

Thanks

**Comment by wobba** [comment]
The query API supports aggregations or filters/refiners as part of the result set. In the search index some managed properties are set as "refinable", allowing the query to return all unique values across all items, not just the ones returned in the page you see.

It might not always be "all" and the count can be incorrect as there are internal optimizations in the search engine to ensure a query will not be too slow. Hope this helps. But it's a "deep refiner", where as a shallow refiner is based on top X results returned.

**Comment by mvcsharepointdev** [comment]
@wobba ok thanks so again if you take this example:-

Let take this as an example. let say we have only one .xls document out of the 3 million documents. and let say this .xls document will be the last document shown inside the Search Result, So will this option ".xls" be shown when the PnP Filter search web part got rendered for the first time, even if this .xls file have not been loaded from the server yet?


should i expect the .xls to be shown inside the Filter FileType combobox? when the filter web part first load? and yes in my case the FileType is refinable ? Thanks again for your help

**Reply by wobba** [reply]
Yes, for most cases it would. But no 100% guarantee. In reality, just don't think about it. It works as expected 🙂
